static/skin/taskbar.js

```diff
@@ -36,15 +37,32 @@ document.addEventListener('DOMContentLoaded', function () {
         noResults: true,
         /* We must display 10 results (requested) + 1 potential link to do a full text search. */
         maxResults: 11,
-        highlight: true
       },
       resultItem: {
         element: (item, data) => {
-          item.innerHTML = `${htmlDecode(data.value.label)}`;
+          let searchLink;
+          if (data.value.kind == "path") {
+            searchLink = `${root}/${bookName}/${htmlDecode(data.value.path)}`;
+          } else {
+            searchLink = `${root}/search?content=${bookName}&pattern=${encodeURIComponent(htmlDecode(data.value.value))}`;
```

Comment:
Now you've undone the fix of https://github.com/kiwix/libkiwix/pull/796#discussion_r933761318

Reply:
Yes, for reason (https://github.com/kiwix/libkiwix/pull/796#discussion_r933978052)
URL encoding `bookName` here will lead to a URL like 
`http://192.168.18.8:8080/zimfile%26other/A/Heaven_Knows_(Taylor_Hicks_song)`
After clicking that suggestion if we search again, the `bookName` that is sent will be:
`zimfile%26other`
This is URL encoded to
`zimfile%2526other`
(`%` turns into `%25`)
Fetch fails, no suggestions.

Reply:
Both in the original comment and now too I was only referring to the usage of `bookName` in `/search?content=${bookName}...`

Reply:
It definitely has to be done. If the `/search` endpoint then puts a URL-encoded book name in the links in the search results, that has to be fixed in the back-end.

Reply:
Alternatively, we may always URL-decode `bookName` in the front-end

Reply:
Ok all the time, I kept thinking it was said for both.
Sending a fixup